docs/scos/dev/front-end-development/202212.0/oryx/building-components/create-component.md

```diff
@@ -0,0 +1,211 @@
+
+export class ProductIdComponent extends ProductMixin(LitElement) {
+  protected pricingService = resolve(PricingService);
+}
+```
+
+You can now use the pricing service API in the component. Service methods always return observables (using [RxJS](https://rxjs.dev/)), so that the service can be lazy loaded and the response can be used by [signals](docs/scos/dev/front-end-development/{{oage.version}}/oryx/architecture/reactivity/signals.html) to update the DOM efficiently.
+
+#### Step 6: Prepare for Server Side Rendering and Hydration
+
+If your application needs to be indexed by crawlers, such as Google Search or Pinterest, it is important that the application is [server side rendered](/docs/scos/dev/front-end-development/{{page.version}}/oryx/architecture/oryx-server-side-rendering.html).
+
+When a component is Server Side Rendered, some of the browser APIs are not available. Most commonly known is the `window` or `document` object. It is important to take this into account when you're implementing custom components.
+
+Oryx renders pages on the server and returns the minimum amount of JavaScript needed. Components should not need any JavaScript initially, but when user start interacting with a component, or when the component needs to reflect a certain application state, additional JavaScript needs to be loaded. Loading the component logic at the client is called _hydration_. Hydration is fairly costly as the component logic is loaded over the network and initialized in the application. Additionally, the component might require to fetch data from a backend api. Oryx therefor tries to avoid hydration or at least delays hydration till it is needed.
+
+As a component developer, you are in charge of configuring the hydration trigger. You can use the `@hydrate` decorator that can take an event or context.
+
+The following example shows how to setup the component to be hydrated when the context is changed:
+
+```ts
+import { resolve } from "@spryker-oryx/di";
+import { ProductMixin } from "@spryker-oryx/product";
+import { hydrate } from "@spryker-oryx/utilities";
+
+@hydrate({ context: ProductContext.SKU })
+export class ProductIdComponent extends ProductMixin(LitElement) {
+  // ...
+}
+```
+
+Alternatively, you can trigger the hydration by a specific event:
+
+```ts
+@hydrate({ event: ["mouseover", "focus"] })
+export class ProductIdComponent extends ProductMixin(LitElement) {
+  // ...
+}
+```
+
+## Component Definition
+
+The component implementation that we start building in the first section of this documentation is not yet registered in an Oryx application. To make the component available to the application, we need to register a [component definition](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-components/component-definition.html) in the application.
+
+```ts
+import { componentDef } from "@spryker-oryx/utilities";
+
+export const productIdComponent = componentDef({
+  name: "oryx-product-id",
```

Comment:
Maybe it's better to not use standard `oryx` namespace, as long as with this example some developers can get it as a best practice and then have some component name conflicts with our component library. I'd suggest to use something neutral, `project-name` / `custom` / etc.